src/test/java/org/apache/commons/imaging/formats/jpeg/exif/ExifRewriterRoundtripTest.java

```diff
@@ -56,124 +61,160 @@ public static Stream<File> data() throws Exception {
+
+                assertTrue(sameValue, "Bytes are different for field: " + sourceField);
             }
         }
     }
 
     private void copyToDuplicateFile(final File sourceFile, final TiffOutputSet duplicateTiffOutputSet) throws IOException, ImagingException, ImagingException {
+
         final ExifRewriter exifRewriter = new ExifRewriter();
+
         duplicateFile = createTempFile();
+
         try (OutputStream duplicateOutputStream = new BufferedOutputStream(new FileOutputStream(duplicateFile))) {
             exifRewriter.updateExifMetadataLossless(sourceFile, duplicateOutputStream, duplicateTiffOutputSet);
         }
     }
 
     private File createTempFile() {
+
         final String tempDir = FileUtils.getTempDirectoryPath();
         final String tempFileName = this.getClass().getName() + "-" + random.nextLong() + ".tmp";
+
         return new File(tempDir, tempFileName);
     }
 
     private TiffOutputSet duplicateTiffOutputSet(final TiffOutputSet sourceTiffOutputSet) throws ImagingException {
-        final TiffOutputSet duplicateTiffOutputSet = new TiffOutputSet();
+
+        final TiffOutputSet duplicateTiffOutputSet = new TiffOutputSet(
+                sourceTiffOutputSet.byteOrder
+        );
+
         for (final TiffOutputDirectory tiffOutputDirectory : sourceTiffOutputSet) {
             duplicateTiffOutputSet.addDirectory(tiffOutputDirectory);
         }
+
         return duplicateTiffOutputSet;
     }
 
-    private JpegImageMetadata getJpegImageMetadata(final File sourceFile) throws ImagingException, IOException {
+    private JpegImageMetadata getJpegImageMetadata(final File sourceFile) throws IOException {
+
         final JpegImageMetadata jpegImageMetadata = (JpegImageMetadata) Imaging.getMetadata(sourceFile);
-        if (null == jpegImageMetadata) {
+
+        if (jpegImageMetadata == null) {
             throw new TestSkippedException();
         }
+
         return jpegImageMetadata;
     }
 
     private TiffImageMetadata getTiffImageMetadata(final JpegImageMetadata sourceJpegImageMetadata) {
+
         final TiffImageMetadata tiffImageMetadata = sourceJpegImageMetadata.getExif();
-        if (null == tiffImageMetadata) {
+
+        if (tiffImageMetadata == null) {
             throw new TestSkippedException();
         }
+
         return tiffImageMetadata;
     }
 
     private TiffOutputSet getTiffOutputSet(final TiffImageMetadata sourceTiffImageMetadata) throws ImagingException {
+
         final TiffOutputSet tiffOutputSet = sourceTiffImageMetadata.getOutputSet();
+
         if (tiffOutputSet == null) {
             throw new TestSkippedException();
         }
+
         return tiffOutputSet;
     }
 
     @AfterEach
     void tearDown() {
+
         if (duplicateFile != null && duplicateFile.exists()) {
             duplicateFile.delete();
             duplicateFile.deleteOnExit();
         }
     }
 
-    @ParameterizedTest
```

Comment:
This `toString()` comparison test is useless.

The TiffWriter does change the order of fields in an attempt to optimize for space. This results in fields being reordered and unnecessary offset fields to be dropped.

You only can compare the `TiffOutputSet`.

I assume the author did this to compare values. This is why I introduced the value comparison API to TiffOutputField.